### ISSUE TITLE
Problem with function phylomatic from package ‘brranching’ - Error in curl::curl_fetch_memory(x$url$url, handle = x$url$handle)

### ISSUE THREAD

**Issue by karlandraczek** [ropensci/brranching#46]
Hi, my name is Karl and I am a PhD in Ecology. 

I am currently doing a plant trait analysis for gymnosperm species and wanted to correct for phylogeny using the function phylomatic.
Unfortunately, I recieved an error message when trying to run: 

> phylo.data <- phylomatic(Data$gymno.phylogeny, taxnames = FALSE, get="POST", storedtree="zanne2014") 
Error in curl::curl_fetch_memory(x$url$url, handle = x$url$handle) : 
  Recv failure: Connection was reset

Here is my original species list:

> Data$gymno.phylogeny
 [1] "cupressaceae/chamaecyparis/Chamaecyparis_obtusa"      
 [2] "cupressaceae/chamaecyparis/Chamaecyparis_pisifera"    
 [3] "cupressaceae/juniperus/Juniperus_chinensis"           
 [4] "cupressaceae/juniperus/Juniperus_monosperma"          
 [5] "cupressaceae/juniperus/Juniperus_sabina"              
 [6] "cupressaceae/metasequoia/Metasequoia_glyptostroboides"
 [7] "cupressaceae/taxodium/Taxodium_distichum"             
 [8] "cupressaceae/thuja/Thuja_occidentalis"                
 [9] "cupressaceae/cunninghamia/Cunninghamia_lanceolata"    
[10] "cupressaceae/cryptomeria/Cryptomeria_japonica"        
[11] "cupressaceae/cupressus/Cupressus_funebris"            
[12] "cupressaceae/platycladus/Platycladus_orientalis"      
[13] "ephedraceae/ephedra/Ephedra_distachya"                
[14] "ginkgoaceae/ginkgo/Ginkgo_biloba"                     
[15] "pinaceae/larix/Larix_decidua"                         
[16] "pinaceae/larix/Larix_gmelinii"                        
[17] "pinaceae/larix/Larix_kaempferi"                       
[18] "pinaceae/picea/Picea_abies"                           
[19] "pinaceae/picea/Picea_glauca"                          
[20] "pinaceae/picea/Picea_koraiensis"                      
[21] "pinaceae/picea/Picea_meyeri"                          
[22] "pinaceae/picea/Picea_rubens"                          
[23] "pinaceae/picea/Picea_wilsonii"                        
[24] "pinaceae/pinus/Pinus_armandii"                        
[25] "pinaceae/pinus/Pinus_edulis"                          
[26] "pinaceae/pinus/Pinus_elliottii"                       
[27] "pinaceae/pinus/Pinus_koraiensis"                      
[28] "pinaceae/pinus/Pinus_massoniana"                      
[29] "pinaceae/pinus/Pinus_palustris"                       
[30] "pinaceae/pinus/Pinus_pumila"                          
[31] "pinaceae/pinus/Pinus_resinosa"                        
[32] "pinaceae/pinus/Pinus_strobus"                         
[33] "pinaceae/pinus/Pinus_sylvestris"                      
[34] "pinaceae/pinus/Pinus_tabuliformis"                    
[35] "pinaceae/pinus/Pinus_taeda"                           
[36] "pinaceae/pinus/Pinus_virginiana"                      
[37] "pinaceae/tsuga/Tsuga_canadensis"                      
[38] "pinaceae/abies/Abies_alba"                            
[39] "pinaceae/abies/Abies_nephrolepis"                     
[40] "podocarpaceae/dacrycarpus/Dacrycarpus_imbricatus"     
[41] "podocarpaceae/podocarpus/Podocarpus_neriifolius"      
[42] "sciadopityaceae/sciadopitys/Sciadopitys_verticillata" 
[43] "taxaceae/cephalotaxus/Cephalotaxus_harringtonii"      
[44] "taxaceae/taxus/Taxus_chinensis"                       
[45] "taxaceae/taxus/Taxus_cuspidata" 

I also tried to set get = "GET", but then R takes ages to run without any result. 

Could anyone help me here? 

Thank you very much for your help!!  

sessionInfo:

R version 4.1.2 (2021-11-01)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19044)

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] taxize_0.9.99    rentrez_1.2.3    brranching_0.7.0 ape_5.6-1        forcats_0.5.1    stringr_1.4.0    dplyr_1.0.7      purrr_0.3.4      readr_2.0.1      tidyr_1.1.3      tibble_3.1.4     ggplot2_3.3.5   
[13] tidyverse_1.3.1  here_1.0.1      

loaded via a namespace (and not attached):
  [1] colorspace_2.0-2        phylocomr_0.3.2         ellipsis_0.3.2          rprojroot_2.0.2         htmlTable_2.4.0         base64enc_0.1-3         fs_1.5.0                httpcode_0.3.0         
  [9] rstudioapi_0.13         urltools_1.7.3          fansi_0.5.0             lubridate_1.7.10        xml2_1.3.2              ritis_1.0.0             codetools_0.2-18        splines_4.1.2          
 [17] mnormt_2.0.2            bold_1.2.0              knitr_1.34              Formula_1.2-4           jsonlite_1.7.2          broom_0.7.9             cluster_2.1.2           dbplyr_2.1.1           
 [25] png_0.1-7               compiler_4.1.2          httr_1.4.2              backports_1.2.1         assertthat_0.2.1        Matrix_1.3-4            fastmap_1.1.0           cli_3.0.1              
 [33] htmltools_0.5.2         tools_4.1.2             igraph_1.2.11           coda_0.19-4             gtable_0.3.0            glue_1.4.2              clusterGeneration_1.3.7 maps_3.4.0             
 [41] fastmatch_1.1-3         Rcpp_1.0.7              cellranger_1.1.0        vctrs_0.3.8             crul_1.2.0              nlme_3.1-153            conditionz_0.1.0        iterators_1.0.14       
 [49] xfun_0.26               rvest_1.0.1             lifecycle_1.0.0         phangorn_2.8.1          XML_3.99-0.8            zoo_1.8-9               MASS_7.3-54             scales_1.1.1           
 [57] hms_1.1.0               parallel_4.1.2          expm_0.999-6            RColorBrewer_1.1-2      curl_4.3.2              gridExtra_2.3           triebeard_0.3.0         rpart_4.1-15           
 [65] reshape_0.8.8           latticeExtra_0.6-29     stringi_1.7.4           foreach_1.5.2           plotrix_3.8-2           checkmate_2.0.0         phytools_1.0-1          rlang_0.4.11           
 [73] pkgconfig_2.0.3         lattice_0.20-45         htmlwidgets_1.5.4       tidyselect_1.1.1        plyr_1.8.6              magrittr_2.0.1          R6_2.5.1                generics_0.1.0         
 [81] Hmisc_4.6-0             combinat_0.0-8          DBI_1.1.1               pillar_1.6.2            haven_2.4.3             foreign_0.8-81          withr_2.4.3             survival_3.2-13        
 [89] scatterplot3d_0.3-41    nnet_7.3-16             modelr_0.1.8            crayon_1.4.1            uuid_0.1-4              utf8_1.2.2              solrium_1.2.0           tmvnsim_1.0-2          
 [97] tzdb_0.1.2              jpeg_0.1-9              grid_4.1.2              readxl_1.3.1            data.table_1.14.0       reprex_2.0.1            digest_0.6.27           numDeriv_2016.8-1.1    
[105] munsell_0.5.0           quadprog_1.5-8

**Comment by rdinnager** [comment]
Same error message for me, using the same arguments (but different taxa obviously).

**Comment by rdinnager** [comment]
It looks like the reason this doesn't work is that the phylomatic web service no longer exists: https://camwebb.info/doc/phylomatic.html . You will have to rely on `phylocom_local()` instead now.

**Comment by karlandraczek** [comment]
Thanks for the hint! Yes, I also found the note that the phylomatic web service was shut down recently. But even when trying phylomatic_local(), it did not work either. Did phylomatic_local() work in your case?

In the end, I used a downloaded gymnosperm tree and matched my species manually using match.phylo.data() from the package picante (https://www.rdocumentation.org/packages/picante/versions/1.8.2). With that you also end up with a phylogenetic object, similar to the ouput using phylomatic() or hylomatic_local().
